tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -111,14 +111,7 @@ bool IsHLSLNumericUserDefinedType(clang::QualType type) {
   const clang::Type *Ty = type.getCanonicalType().getTypePtr();
   if (const RecordType *RT = dyn_cast<RecordType>(Ty)) {
     const RecordDecl *RD = RT->getDecl();
-    if (isa<ClassTemplateSpecializationDecl>(RD)) {
-      return false;   // UDT are not templates
-    }
-    // TODO: avoid check by name
-    StringRef name = RD->getName();
-    if (name == "ByteAddressBuffer" ||
-        name == "RWByteAddressBuffer" ||
-        name == "RaytracingAccelerationStructure")
+    if (!IsUserDefinedRecordType(type))
```

Comment:
This will exclude use of `BuiltInTriangleIntersectionAttributes`, which should return true for `IsHLSLNumericOrAggregateOfNumericType`.

Reply:
Will this also exclude built-in matrix or vector types?  That would not be good.

Reply:
I think excluding matrix and vector types here is what we want because we exclude those in some places since they can't be annotated. If the use case wants to allow matrix and vectors, you call `IsHLSLNumericOrAggregateOfNumericType` which will return true for matrix and vector types. If you want to disallow those you call `IsHLSLNumericUserDefinedType`, which returns false for those cases.

Reply:
Right, I think the vector/matrix were excluded by being template specializations before.